plugins/kubernetes/app/models/kubernetes/role.rb

```diff
@@ -24,13 +24,18 @@ class Role < ActiveRecord::Base
     # create initial roles for a project by reading kubernetes/*{.yml,.yaml,json} files into roles
     def self.seed!(project, git_ref)
       kubernetes_config_files_in_repo(project, git_ref).each do |config_file|
+        scope = where(project: project)
+        next if scope.where(config_file: config_file.file_path).exists?
+
         service_name = config_file.service && config_file.service.metadata.name
-        service_name << "-#{rand(9999999)}" if where(service_name: service_name).exists?
+        service_name << "-#{rand(9999999)}" if service_name && scope.where(service_name: service_name).exists?
```

Comment:
I think this is fine for now, and create a story to investigate that behavior later?